### DISCUSSION TITLE
Strapi admin panel

### DISCUSSION THREAD

**Discussion by jattoo** [strapi/strapi#7305] [Legacy Topics]
Hi, 
I have been recently testing out strapi. However i ran into some issues yesterday and cant seem to understand what really going on. I am asking this question here since most users ran into a statusCode 403 unlike mine which returned 404 (not found).   Everything was working fine until i recently installed stripe 8.81.0. 
At the moment when i do `strapi develop/start` , the admin on the console returns 200 but however when i try to go there through the browser i get a `user-permissions/init 404`. I tried uninstalling and reinstalling the strapi-plugin-users-permissions but no changes. 

I am on:
`strapi 3.1.3
node 12.18.0
npm 6.14.4
mongodb cloud atlas
`
console output if it helps to 
`Welcome back!
To manage your project �, go to the administration panel at:
http://localhost:1337/admin

To access the server ⚡️, go to:
http://localhost:1337

[2020-08-03T17:36:41.084Z] debug GET /admin (47 ms) 200
[2020-08-03T17:36:41.908Z] debug GET /admin/runtime~main.07da870a.js (30 ms) 200
[2020-08-03T17:36:41.908Z] debug GET /admin/main.b045701d.chunk.js (2 ms) 200
[2020-08-03T17:36:41.084Z] debug GET /admin (47 ms) 200
[2020-08-03T17:36:41.908Z] debug GET /admin/runtime~main.07da870a.js (30 ms) 200
[2020-08-03T17:36:41.908Z] debug GET /admin/main.b045701d.chunk.js (2 ms) 200
[2020-08-03T17:36:42.986Z] debug GET /users-permissions/init (4 ms) 404
[2020-08-03T17:36:52.245Z] debug GET index.html (115 ms) 200
[2020-08-03T17:36:52.822Z] debug GET /assets/images/logo_login.png (326 ms) 200
[2020-08-03T17:36:56.190Z] debug GET index.html (87 ms) 200
[2020-08-03T17:36:56.407Z] debug GET /favicon.ico (20 ms) 200
[2020-08-03T17:37:00.069Z] debug GET /admin (3 ms) 200
[2020-08-03T17:37:01.121Z] debug GET /users-permissions/init (8 ms) 404`

However the intresting scenario here is that this however have not stop my frontend from working with backend. I am not simply just able to work in the strapi admin panel since it will not allow me in login. Any suggestions is highly welcome. Thanks guys! you are the best!

**Comment by derrickmehaffy** [comment]
Can you tell me which database you are using? (if you used `--quickstart` it would be SQLite).

If possible can you also paste your `package.json` here inside of a code block?

**Comment by jattoo** [comment]
I am using MongoDBAtlas (cloud). 
package.json:
```
{
  "name": "server",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "devDependencies": {},
  "dependencies": {
    "@popperjs/core": "^2.4.4",
    "jquery": "^3.5.1",
    "koa": "^2.13.0",
    "strapi": "^3.1.3",
    "strapi-admin": "^3.1.3",
    "strapi-connector-mongoose": "^3.1.3",
    "strapi-plugin-content-manager": "^3.1.3",
    "strapi-plugin-content-type-builder": "^3.1.3",
    "strapi-plugin-email": "^3.1.3",
    "strapi-plugin-graphql": "^3.1.3",
    "strapi-plugin-upload": "^3.1.3",
    "strapi-plugin-users-permissions": "^3.1.3",
    "strapi-utils": "^3.1.3",
    "stripe": "^8.81.0"
  },
  "author": {
    "name": "moe"
  },
  "strapi": {
    "uuid": "a32eb50b-9cdf-472a-b097-a2d117eeb3af"
  },
  "engines": {
    "node": ">=10.0.0",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}
```

**Reply by derrickmehaffy** [reply]
It shouldn't be causing the issue here, but can you please remove all the `^` before the Strapi packages and you shouldn't need to define `koa` as a depends here, so something like this:

```
{
  "name": "server",
  "private": true,
  "version": "0.1.0",
  "description": "A Strapi application",
  "scripts": {
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "devDependencies": {},
  "dependencies": {
    "@popperjs/core": "^2.4.4",
    "jquery": "^3.5.1",
    "strapi": "3.1.3",
    "strapi-admin": "3.1.3",
    "strapi-connector-mongoose": "3.1.3",
    "strapi-plugin-content-manager": "3.1.3",
    "strapi-plugin-content-type-builder": "3.1.3",
    "strapi-plugin-email": "3.1.3",
    "strapi-plugin-graphql": "3.1.3",
    "strapi-plugin-upload": "3.1.3",
    "strapi-plugin-users-permissions": "3.1.3",
    "strapi-utils": "3.1.3",
    "stripe": "^8.81.0"
  },
  "author": {
    "name": "moe"
  },
  "strapi": {
    "uuid": "a32eb50b-9cdf-472a-b097-a2d117eeb3af"
  },
  "engines": {
    "node": ">=10.0.0",
    "npm": ">=6.0.0"
  },
  "license": "MIT"
}
```

You should clear out your: 
- `node_modules`
- `yarn.lock` (if exists)
- `package-lock.json` (if exists)

then rerun either a `yarn install` or `npm install` and rebuild the admin using `yarn build` or `npm run build`.

If there is still issues, we can do a bit more debugging.

**Reply by jattoo** [reply]
@derrickmehaffy : Your steps you provided solved my issue. However it seems the admin panel is now enforcing login with an email address instead of a username. Is there something i can do about this since without possibility to login with username, it seems i would be locked out.

**Reply by derrickmehaffy** [reply]
I believe this was a change in the 3.1.x which no longer requires a username during the Admin registration process. I'll bring this up with the team to see if it can still support username login (if the username is set) or if it doesn't, if we could add that back as an option.

**Reply by derrickmehaffy** [reply]
Just as a reference too @jattoo I'm guessing the `^` was added because you used something like `npm update` which we don't generally recommend as there are sometimes migration guides (which there was from v3.0.x to v3.1.x: https://strapi.io/documentation/v3.x/migration-guide/migration-guide-3.0.x-to-3.1.x.html ) before you update versions, I'd suggest you check to see if there is a new guide, they are always linked on the release if needed such as here: https://github.com/strapi/strapi/releases/tag/v3.1.0

**Reply by jattoo** [reply]
I did some `npm update ` and `npm audit fix` whilst i was trying to seek ways of solving the issue. That must have been the case. I took off all the carets already.  Will it help if i add email to the schema in mongodb? Will it allow to login with the email as same user even if it was not there during the original registration.  If it is impossible to login with the username , I might have to go blind on this because i already have so much in the backend and cant simply clean all to be able to work with the admin panel. Is there some guide on doing stuff on the terminal especially whilst working with content type builder?

**Reply by derrickmehaffy** [reply]
> Is there some guide on doing stuff on the terminal especially whilst working with content type builder?

There the `strapi generate:api` (among other generators) but nothing that supports relations, it's only very basic model generation for normal fields at the moment.

**Reply by derrickmehaffy** [reply]
@jattoo I did check with the team and the username signin for the admin panel has been removed as that field is no longer required and is no longer unique (only the email).

This is to allow for some SSO stuff coming later on in v4, it was a requirement to get rid of the unique constraint for the username.

**Reply by jattoo** [reply]
@derrickmehaffy ...i guess that's it then. Atleast  now i know its best to register with email instead. Thnx bro.

**Reply by derrickmehaffy** [reply]
No problem, I think it will be more clear as to the reasoning in the future, I did talk with the team and will have them implement a more generic changelog in addition to the typical migration guide to provide more information on these changes (as neither me nor @lauriejim were aware of this change)
